### PR TITLE
B2B-1237: [update] sync recent updates from B2B Edition API. 

### PR DESCRIPTION
**Pull request by JamessenLiu**
# [B2B-1327](https://bigcommercecloud.atlassian.net/browse/B2B-1327)

## What changed?

* add tag for channel endpoints. Because the document did not display correctly when the tag was missing.
image

* sync independent comapny docs from B2B stoplight
    *  [Jira ticket: B2B-1201](https://bigcommercecloud.atlassian.net/browse/B2B-1201)
    * [origin pr](https://github.com/B3BC/boson/pull/464)

* remove task section
    * [Jira ticket: B2B-1222](https://bigcommercecloud.atlassian.net/browse/B2B-1222) 
    * [origin pr](https://github.com/B3BC/boson/pull/463)

* add minModified query param for sales staff api
    * [origin pr](https://github.com/B3BC/boson/pull/456)
    * [Jira ticket: BUN-2720](https://bigcommercecloud.atlassian.net/browse/BUN-2720) 

## Release notes draft

N/A

## Anything else?

ping {names}

[B2B-1327]: https://bigcommercecloud.atlassian.net/browse/B2B-1327?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 1]
```diff
@@ -448,7 +448,10 @@ paths:
       tags:
         - Company
       summary: Create a Company
-      description: Create a company
+      description: |-
+        Create a company.
+        * When the `independent company behavior` is enabled, the system will not automatically create a dedicated customer group for each new Company account. Instead, you can assign a customer group to a Company account as needed, using the `customerGroupId` field. If the `customerGroupId` field is not provided, the B2B company will be associated with the default customer group you have configured. If no default customer group is configured, the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to No Group in BigCommerce.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:453]:
```suggestion
        * When the `independent company behavior` is enabled, the system will not automatically create a dedicated customer group for each new Company account. Instead, you can assign a customer group to a Company account as needed using the `customerGroupId` field. If the `customerGroupId` field is not provided, the B2B company will be associated with your configured default customer group. If you have not configured a default customer group, the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to "No Group" in BigCommerce.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 2]
```diff
@@ -448,7 +448,10 @@ paths:
...
+        * When the `independent company behavior` is disabled, a new customer group will be automatically created and associated with each new Company account. In this case, the `customerGroupId` field will not be used.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:454]:
```suggestion
        * When the `independent company behavior` is disabled, a new customer group will be automatically created and associated with each new Company account. In this case, you will not use the `customerGroupId` field.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 3]
```diff
@@ -531,6 +534,9 @@ paths:
                     originChannelId:
                       type: integer
                       description: "BigCommerce channel ID, used for BigCommerce customer origin channel ID. This field takes effect only when you do not configure the store default B2B channel."
+                    customerGroupId:
+                      type: integer
+                      description: BigCommerce Customer Group ID. Used to associate a BigCommerce Customer Group to a B2B company when the  independent company behavior is enabled.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:539]:
```suggestion
                      description: BigCommerce Customer Group ID. Used to associate a BigCommerce Customer Group to a B2B company when the independent company behavior is enabled.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 4]
```diff
@@ -866,7 +872,10 @@ paths:
       tags:
         - Company
       summary: Update a Company
-      description: Update a company's attributes
+      description: |
+        Update a company's attributes.
+        * When the `independent company behavior` is enabled, you can assign a customer group to a Company using the `customerGroupId` field. Once assigned, the customer group will be associated with all users within that Company. If `customerGroupId` is set to 0, the customer group association will be removed from the Company, and all users within the Company will no longer have a group assigned. The `priceListAssign` field will not take effect; please configure the price list for the customer group directly in the BigCommerce Control Panel.
+        * When the `independent company behavior` is disabled, the `customerGroupId` field will not take effect. The association between B2B companies and BigCommerce customer groups cannot be changed. The `priceListAssign` field will be used to configure the price list assigned to the Company.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:878]:
```suggestion
        * When the `independent company behavior` is disabled, the `customerGroupId` field will not take effect. You cannot change the association between B2B companies and BigCommerce customer groups. You will use the `priceListAssign` field to configure the price list assigned to the Company.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 5]
```diff
@@ -1029,7 +1042,10 @@ paths:
       tags:
         - Company
       summary: Delete a Company
-      description: Delete A Company
+      description: |-
+        Delete A Company.
+        * When the `independent company behavior` is disabled. Deleting a company will also delete the corresponding BigCommerce Customer Group.
+        * When the `independent company behavior` is enabled. Deleting a company will not affect the customer group associated with the company
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1048]:
```suggestion
        * When the `independent company behavior` is enabled. Deleting a company will not affect the customer group associated with the company.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 6]
```diff
@@ -1112,7 +1128,10 @@ paths:
       tags:
         - Company
       summary: Convert from BigCommerce CustomerGroup into Company
-      description: Create company from bigCommerce customer group. This API is disabled for MSF store
+      description: |-
+        Create company from bigCommerce customer group. 
+        * This API is disabled for MSF store
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1133]:
```suggestion
        * This API is turned off for the MSF store.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 7]
```diff
@@ -1112,7 +1128,10 @@ paths:
...
+        * This API is disabled when independent company feature enabled
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1134]:
```suggestion
        * This API is turned off when the independent company feature is enabled.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 8]
```diff
@@ -1112,7 +1128,10 @@ paths:
       tags:
         - Company
       summary: Convert from BigCommerce CustomerGroup into Company
-      description: Create company from bigCommerce customer group. This API is disabled for MSF store
+      description: |-
+        Create company from bigCommerce customer group.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1132]:
```suggestion
        Create company from a BigCommerce customer group. 
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 9]
```diff
@@ -1577,7 +1596,10 @@ paths:
                           _detail: string
                     meta:
                       message: string
-      description: Bulk create companies
+      description: |-
+        Bulk create companies
+        * When the `independent company behavior` is enabled, the system will not automatically create a dedicated customer group for each new Company account. Instead, you can assign a customer group to a Company account as needed, using the `customerGroupId` field. If the `customerGroupId` field is not provided, the B2B company will be associated with the default customer group you have configured. If no default customer group is configured, the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to No Group in BigCommerce.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1601]:
```suggestion
        * When the `independent company behavior` is enabled, the system will not automatically create a dedicated customer group for each new Company account. Instead, you can assign a customer group to a Company account as needed using the `customerGroupId` field. If the `customerGroupId` field is not provided, the B2B company will be associated with the configured default customer group. If you have not configured a default customer group, the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to "No Group" in BigCommerce.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 10]
```diff
@@ -1577,7 +1596,10 @@ paths:
...
+        * When the `independent company behavior` is disabled, a new customer group will be automatically created and associated with each new Company account. In this case, the `customerGroupId` field will not be used.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1602]:
```suggestion
        * When the `independent company behavior` is turned off, a new customer group will be automatically created and associated with each new Company account. In this case, you will not use the `customerGroupId` field.
```

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 11]
```diff
@@ -1641,7 +1667,10 @@ paths:
       tags:
         - Company
       summary: Update Companies (Batch)
-      description: Batch update companies
+      description: |
+        Batch update companies.
+        * When the `independent company behavior` is enabled, you can assign a customer group to a Company using the `customerGroupId` field. Once assigned, the customer group will be associated with all users within that Company. If `customerGroupId` is set to 0, the customer group association will be removed from the Company, and all users within the Company will no longer have a group assigned. The `priceListAssign` field will not take effect; please configure the price list for the customer group directly in the BigCommerce Control Panel.
+        * When the `independent company behavior` is disabled, the `customerGroupId` field will not take effect. The association between B2B companies and BigCommerce customer groups cannot be changed. The `priceListAssign` field will be used to configure the price list assigned to the Company.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1673]:
```suggestion
        * When the `independent company behavior` is turned off, the `customerGroupId` field will not take effect. You cannot change the association between B2B companies and BigCommerce customer groups. You will use the `priceListAssign` field to configure the price list assigned to the Company.
```

File: docs/b2b-edition/specs/api-v3/user/user.yaml [hunk 12]
```diff
@@ -355,7 +355,10 @@ paths:
       tags:
         - User
       summary: Create a User
-      description: 'Create a company user, which belongs company'
+      description: |-
+        Create a company user within an existing company. 
+        * When the `independent company behavior` is enabled, B2B Edition company accounts are the source of truth for defining a company user’s customer group assignment. If no default customer group is configured or set value of `customerGroupId` to 0,  the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to No Group in BigCommerce
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/user/user.yaml:360]:
```suggestion
        * When the `independent company behavior` is enabled, B2B Edition company accounts are the source of truth for defining a company user’s customer group assignment. If you do not configure a default customer group or the value of `customerGroupId` is 0,  the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to "No Group" in BigCommerce.
```

File: docs/b2b-edition/specs/api-v3/user/user.yaml [hunk 13]
```diff
@@ -355,7 +355,10 @@ paths:
...
+        * When the `independent company behavior` is disabled, the BigCommerce customer record’s customer group assignment will be the source of truth for defining a company user’s company assignment.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/user/user.yaml:361]:
```suggestion
        * When the `independent company behavior` is turned off, the BigCommerce customer record’s customer group assignment will be the source of truth for defining a company user’s company assignment.
```

File: docs/b2b-edition/specs/api-v3/user/user.yaml [hunk 14]
```diff
@@ -507,6 +510,8 @@ paths:
       description: |-
         Create company users in batch. All of the user should be in same 
          company.
+         * When the `independent company behavior` is enabled, B2B Edition company accounts are the source of truth for defining a company user’s customer group assignment. If no default customer group is configured or set value of `customerGroupId` to 0,  the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to No Group in BigCommerce
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/user/user.yaml:513]:
```suggestion
         * When the `independent company behavior` is enabled, B2B Edition company accounts are the source of truth for defining a company user’s customer group assignment. If you do not configure a default customer group or the value of `customerGroupId` is 0,  the B2B company will not be associated with any customer group, and the company users within the company will have their corresponding customer record assigned to "No Group" in BigCommerce.
```

File: docs/b2b-edition/specs/api-v3/user/user.yaml [hunk 15]
```diff
@@ -507,6 +510,8 @@ paths:
...
+        * When the `independent company behavior` is disabled, the BigCommerce customer record’s customer group assignment will be the source of truth for defining a company user’s company assignment.
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/user/user.yaml:514]:
```suggestion
        * When the `independent company behavior` is turned off, the BigCommerce customer record’s customer group assignment will be the source of truth for defining a company user’s company assignment.
```

**Review by bc-traciporter — COMMENTED** [review]
Made comments to change to active voice. I tried to use the term "disabled" a little less. Some readers may consider the use of "disabled" as insensitive.

**Comment by JamessenLiu** [comment]
@bc-traciporter 
hank you for your review; I have modified the code according to your suggestions.

**Review by bc-traciporter — APPROVED** [review]
Approved

**Comment by bc-traciporter** [comment]
@JamessenLiu Would you like for me to merge this PR?

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 16]
```diff
@@ -1056,7 +1072,10 @@ paths:
       tags:
         - Company
       summary: Convert from BigCommerce CustomerGroup into Company
-      description: Create company from bigCommerce customer group. This API is disabled for MSF store
+      description: |-
+        Create company from a BigCommerce customer group. 
+        * This API is turned off for the MSF store.
```
Comment (becomevocal) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:1077]: Is the API turned off, or this _endpoint_ not available? And does the endpoint not work at all, or does it return an error?

Comment (JamessenLiu) [reply docs/b2b-edition/specs/api-v3/company/company.yaml:1077]: If the store disable the B2B MSF feature, this API is avaliable; 
If the store enable the B2B MSF feature, this API is not available, and will return an error .

**Review by becomevocal — APPROVED** [review]

**Pull request merged**